lib/rubocop/cop/rspec/sort_metadata.rb

```diff
@@ -0,0 +1,94 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module RSpec
+      # Sort RSpec metadata alphabetically.
+      #
+      # @example
+      #   # bad
+      #   describe 'Something', :b, :a
+      #   context 'Something', foo: 'bar', baz: true
+      #   it 'works', :b, :a, foo: 'bar', baz: true
+      #
+      #   # good
+      #   describe 'Something', :a, :b
+      #   context 'Something', baz: true, foo: 'bar'
+      #   it 'works', :a, :b, baz: true, foo: 'bar'
+      #
+      class SortMetadata < Base
+        extend AutoCorrector
+        include RangeHelp
+
+        MSG = 'Sort metadata alphabetically.'
+
+        # @!method rspec_metadata(node)
+        def_node_matcher :rspec_metadata, <<~PATTERN
+          (send #rspec? {#Examples.all #ExampleGroups.all #SharedGroups.all #Hooks.all #Includes.all} _ $sym* (hash $...)?)
+        PATTERN
+
+        # @!method rspec_configure(node)
+        def_node_matcher :rspec_configure, <<~PATTERN
+          (block (send #rspec? :configure) (args (arg $_)) ...)
+        PATTERN
+
+        # @!method metadata_in_block(node)
+        def_node_search :metadata_in_block, <<~PATTERN
+          (send (lvar $_) #Hooks.all _ $sym* (hash $...)?)
+        PATTERN
+
+        def on_block(node)
+          rspec_configure(node) do |block_var|
+            metadata_in_block(node).each do |receiver, symbols, pairs|
+              investigate(symbols, pairs.flatten) if receiver == block_var
+            end
+          end
+        end
+
+        def on_send(node)
+          rspec_metadata(node) do |symbols, pairs|
+            investigate(symbols, pairs.flatten)
+          end
+        end
+
+        private
+
+        def investigate(symbols, pairs)
+          return if sorted?(symbols, pairs)
+          return unless (crime_scene = crime_scene(symbols, pairs))
+
+          add_offense(crime_scene) do |corrector|
+            corrector.replace(crime_scene, replacement(symbols, pairs))
+          end
+        end
+
+        def crime_scene(symbols, pairs)
+          metadata = symbols + pairs
+
+          range = range_between(
+            metadata.first.loc.expression.begin_pos,
+            metadata.last.loc.expression.end_pos
+          )
+
+          range if range.last_line == range.first_line
+        end
+
+        def replacement(symbols, pairs)
+          (sort_symbols(symbols) + sort_pairs(pairs)).map(&:source).join(', ')
+        end
+
+        def sorted?(symbols, pairs)
+          symbols == sort_symbols(symbols) && pairs == sort_pairs(pairs)
+        end
+
+        def sort_pairs(pairs)
+          pairs.sort_by { |pair| pair.key.value.to_s }
+        end
+
+        def sort_symbols(symbols)
+          symbols.sort_by(&:value)
```

Comment:
To my best knowledge, in theory, keys can be strings, too. And this would fail comparing symbols to strings.
However, this doesn't seem to manifest on several big codebases.